app/src/main/java/com/example/playme/FBReg.kt

```diff
@@ -0,0 +1,160 @@
+                )
+            )
+
+            Spacer(modifier = Modifier.height(16.dp))
+
+            TextField(
+                value = password,
+                onValueChange = { password = it },
+                label = { Text("Пароль") },
+                visualTransformation = PasswordVisualTransformation(),
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(16.dp),
+                colors = TextFieldDefaults.textFieldColors(
+                    textColor = Color.White,
+                    focusedIndicatorColor = Color.White,
+                    unfocusedIndicatorColor = Color.White
+                ),
+                textStyle = MaterialTheme.typography.bodySmall.copy(
+                    fontFamily = customFontFamily,
+                    fontSize = 16.sp,
+                    color = Color.DarkGray
+                )
+            )
+
+            Spacer(modifier = Modifier.height(16.dp))
+
+            TextField(
+                value = confirmPassword,
+                onValueChange = { confirmPassword = it },
+                label = { Text("Підтвердіть пароль") },
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(16.dp),
+                colors = TextFieldDefaults.textFieldColors(
+                    textColor = Color.White,
+                    focusedIndicatorColor = Color.White,
+                    unfocusedIndicatorColor = Color.White
+                ),
+                textStyle = MaterialTheme.typography.bodySmall.copy(
+                    fontFamily = customFontFamily,
+                    fontSize = 16.sp,
+                    color = Color.DarkGray
+                )
+            )
+
+            Spacer(modifier = Modifier.height(16.dp))
+
+            Button(
+                onClick = { registerWithEmail(context, auth, email, password, confirmPassword) },
+                modifier = Modifier.align(Alignment.End),
+            ) {
+                Text(
+                    text = "Зареєструватися",
+                    color = DarkBlue,
+                    fontFamily = customFontFamily,
+                    fontWeight = FontWeight.Bold
+                )
+            }
+        }
+    }
+}
+
+fun registerWithEmail(
+    context: Context,
+    auth: FirebaseAuth,
+    email: String,
+    password: String,
+    confirmPassword: String
+) {
+    if (password == confirmPassword) {
+        auth.createUserWithEmailAndPassword(email, password)
+            .addOnCompleteListener { task ->
+                if (task.isSuccessful) {
+                    Toast.makeText(context, "Реєстрація успішна!", Toast.LENGTH_SHORT).show()
+                } else {
+                    Toast.makeText(context, "Помилка реєстрації: ${task.exception?.message}", Toast.LENGTH_SHORT).show()
+                }
+            }
+    } else {
+        Toast.makeText(context, "Паролі не співпадають", Toast.LENGTH_SHORT).show()
+    }
+}
```

Comment:
Better cut to different file which will relate to Authentication and use as dependency injection